plugins/inputs/mcrouter/mcrouter.go

```diff
@@ -0,0 +1,224 @@
+	"destination_max_pending_reqs",
+	"destination_max_inflight_reqs",
+	"retrans_per_kbyte_max",
+	"cmd_get_count",
+	"cmd_delete_out",
+	"cmd_lease_get",
+	"cmd_set",
+	"cmd_get_out_all",
+	"cmd_get_out",
+	"cmd_lease_set_count",
+	"cmd_other_out_all",
+	"cmd_lease_get_out",
+	"cmd_set_count",
+	"cmd_lease_set_out",
+	"cmd_delete_count",
+	"cmd_other",
+	"cmd_delete",
+	"cmd_get",
+	"cmd_lease_set",
+	"cmd_set_out",
+	"cmd_lease_get_count",
+	"cmd_other_out",
+	"cmd_lease_get_out_all",
+	"cmd_set_out_all",
+	"cmd_other_count",
+	"cmd_delete_out_all",
+	"cmd_lease_set_out_all",
+}
+
+// SampleConfig returns sample configuration message
+func (m *Mcrouter) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns description of Mcrouter plugin
+func (m *Mcrouter) Description() string {
+	return "Read metrics from one or many mcrouter servers"
+}
+
+// Gather reads stats from all configured servers accumulates stats
+func (m *Mcrouter) Gather(acc telegraf.Accumulator) error {
+	if len(m.Servers) == 0 && len(m.UnixSockets) == 0 {
+		return m.gatherServer(":11211", false, acc)
+	}
+
+	for _, serverAddress := range m.Servers {
+		acc.AddError(m.gatherServer(serverAddress, false, acc))
+	}
+
+	for _, unixAddress := range m.UnixSockets {
+		acc.AddError(m.gatherServer(unixAddress, true, acc))
+	}
+
+	return nil
+}
+
+func (m *Mcrouter) gatherServer(
+	address string,
+	unix bool,
+	acc telegraf.Accumulator,
+) error {
+	var conn net.Conn
+	var err error
+	if unix {
+		conn, err = net.DialTimeout("unix", address, defaultTimeout)
+		if err != nil {
+			return err
+		}
+		defer conn.Close()
+	} else {
+		_, _, err = net.SplitHostPort(address)
+		if err != nil {
+			address = address + ":11211"
+		}
+
+		conn, err = net.DialTimeout("tcp", address, defaultTimeout)
+		if err != nil {
+			return err
+		}
+		defer conn.Close()
+	}
+
+	if conn == nil {
+		return fmt.Errorf("Failed to create net connection")
+	}
+
+	// Extend connection
+	conn.SetDeadline(time.Now().Add(defaultTimeout))
+
+	// Read and write buffer
+	rw := bufio.NewReadWriter(bufio.NewReader(conn), bufio.NewWriter(conn))
+
+	// Send command
+	if _, err := fmt.Fprint(rw, "stats\r\n"); err != nil {
+		return err
+	}
+	if err := rw.Flush(); err != nil {
+		return err
+	}
+
+	values, err := parseResponse(rw.Reader)
+	if err != nil {
+		return err
+	}
+
+	// Add server address as a tag
+	tags := map[string]string{"server": address}
+
+	// Process values
+	fields := make(map[string]interface{})
+	for _, key := range sendMetrics {
+		if value, ok := values[key]; ok {
+			// Mostly it is the number
+			if iValue, errParse := strconv.ParseInt(value, 10, 64); errParse == nil {
+				fields[key] = iValue
+			} else if fValue, errParse := strconv.ParseFloat(value, 64); errParse == nil {
+				fields[key] = fValue
+			} else {
+				fields[key] = value
+			}
+		}
+	}
+	acc.AddFields("mcrouter", fields, tags)
+	return nil
+}
+
+func parseResponse(r *bufio.Reader) (map[string]string, error) {
```

Comment:
We should use a `bufio.Scanner` to read lines, the ReadLine() function can return partial lines.

Reply:
Using `bufio.Scanner` now